pkg/server/server.go

```diff
@@ -153,6 +156,22 @@ func (pm *PendingDialManager) Remove(random int64) *ProxyClientConnection {
 	return pd
 }
 
+func (pm *PendingDialManager) RemoveForStream(uid string) []*ProxyClientConnection {
```

Comment:
nit: let's stick with `streamUID`
```suggestion
func (pm *PendingDialManager) RemoveForStream(streamUID string) []*ProxyClientConnection {
```